modules/ROOT/pages/configuring-properties.adoc

```diff
@@ -229,8 +229,9 @@ Configuration properties can be overwritten. The Hierarchy in which these are tr
 . System Properties
 . Environment Properties
 . Deployment Properties
-. Application Properties
+. Application Properties (includes Configuration Properties, Secure Configuration Properties and other Custom Configuration Properties)
+. Global Properties
```

Comment:
Please change the Hierarchies section to this text.

Configuration properties can be overwritten. The system evaluates these properties in the following order:
. System properties
. Environment properties
. Deployment properties
. Application properties
. Global properties

So, for example, if a configuration property named 'size' is defined as both a system property and an application property, the value of the application property named 'size' will override that of the system property named 'size'.

Also, the value of an application property can depend on the values of properties that are higher up int the hierarchy. The value of application properties  can derive from deployment, enviornment, and/or system properties. The value of deployment properties can derive from environment and/or system properties, and so on. For example, if there is a system property named 'env', a configuration property can have the value 'file.${env}.xml'.

Reply:
> the value of the application property named 'size' will override that of the system property named 'size'
This is the opposite of what actually happens, as stated here:
> the value for that application is the one with the most precedence (in this case, the System Property).

Reply:
Properties Hierarchy
Configuration properties can be overwritten. Mule uses the following hierarchy to determine which properties take precedence if they have the same name. In this hierarchy, system properties are at the top, so they take precedence over the other types of properties.

System properties

Environment properties

Deployment properties

Application properties (includes configuration properties, secure configuration properties and other custom configuration properties)

Global Properties

So, for example, if a configuration property named 'size" is defined as a system property, and there is also an application configuration property named 'size', the value for the application is the property with the most precedence (in this case, the system property).

Also, a property can derive its value from properties that are higher in the hierarchy. Therefore, an application property can derive its value from environment, system and/or deployment properties. A deployment property can derive its value from an environment and/or system property, and so on. For example, if there is a system property named 'env', a configuration property can have the value 'file.$github.xml'.